### PR TITLE
plots: remove legacy code from verify_metrics

### PR DESCRIPTION
**Pull request by ykasimov**
* [ x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Fixes: https://github.com/iterative/dvc/issues/7988

### CODE REVIEW

File: dvc/output.py [hunk 1]
```diff
@@ -708,14 +708,13 @@ def verify_metric(self):
...
-
-        if not self.metric or self.plot:
+        if not self.metric:
             return
 
         if not os.path.exists(self.fs_path):
             return
 
-        name = "metrics" if self.metric else "plot"
+        name = "metrics"
```
Comment (pared) [inline dvc/output.py:717]: Now that there is only one choice we can just hardcode it in messages.

Comment (pared) [inline dvc/output.py:717]: Now that there is only one choice we can just hardcode it in messages.

**Review by pared — CHANGES_REQUESTED** [review]

**Review by pared — APPROVED** [review]

File: dvc/output.py [hunk 2]
```diff
@@ -708,22 +708,20 @@ def verify_metric(self):
...
             return
 
         if not os.path.exists(self.fs_path):
             return
 
-        name = "metrics" if self.metric else "plot"
         if os.path.isdir(self.fs_path):
-            msg = "directory '%s' cannot be used as %s."
-            logger.debug(msg, str(self), name)
+            logger.debug(f"directory '{self}' cannot be used as metrics.")
```
Comment (alexmojaki) [inline dvc/output.py:718]: I worked hard on https://github.com/iterative/dvc/pull/7961, don't start undoing it!

Comment (pared) [reply dvc/output.py:718]: Ah, sorry @alexmojaki, I missed that one. 
@ykasimov can you roll the log calls back?

Comment (pared) [reply dvc/output.py:718]: reverted in #8081

**Pull request merged**